### ISSUE TITLE
Version ranges

### ISSUE THREAD

**Issue by nickalbrecht** [dotnet-outdated/dotnet-outdated#68]
TLDR;
Tool doesn't cover version ranges such as `[3.*,4.*)`, which are considered valid versions on the [Nuget PackageReference Documentation](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#examples)

Ran into this today when I realized one of the licensed third party libraries we use got a major update. For example, our license only covers version 3.x but not the newer 4.x one. So to prevent the new version from showing up as an available upgrade in both Visual Studio and from this tool via command line, I added a version range to exclude showing version 4.x. Visual Studio worked fine, but the `dotnet outdated` command gets an error saying that it was unable to process the project, stating that
> '[3.\*,4.\*)' is not a valid version string

**Comment by nickalbrecht** [comment]
I may have been premature in my efforts. The `Manage Nuget Packages` in Visual Studio seemed to accept the version restriction, but that was about it. My build nor the restore packages commands worked with that version specified. The [documentation on consuming nuget packages](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files) via the PackageReference seems to differ from the link I initially was using, which is confusing. I'm honestly unsure if I even can instruct my project (in Visual Studio, or via the outdated tool) to not suggest upgrading to a version we don't have a license for.

**Comment by rajbos** [comment]
NuGet [switched](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges) to full semantic versioning, so the setup for this would now be `[3,4)`. This concept works with dotnet-outdated in the same way Visual Studio does it (unfortunately): 

We have a reference setup to SignalR and our frontend is not ready for the 5.0 version, so we use version ranges to prevent updates.
```xml
<PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="[3,4)" />
```

Both Visual Studio and `dotnet outdated` see this as a potential update to 3.1.16 (current latest version 3). Selecting a version in Visual Studio or running `dotnet outdated -u will now update the line to:

```xml
<PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="3.1.16" />
```

Thus overwriting the given range!
Next time we check for updates it will still push it over to the latest 5.0 version :-(.
